### PR TITLE
Enhancement/4342 analytics step mode

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #4342 

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: .github/workflows/php-tests-wp-4-7-php-5-6.yml [hunk 1]
```diff
@@ -71,7 +71,7 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-composer-
       - name: Composer Install
-        run: docker run --rm -v "$PWD:/app" -v "${{ steps.composer-cache.outputs.dir }}:/tmp/cache" composer install --no-interaction --no-progress # Run inside docker so composer install doesn't error...
+        run: docker run --rm -v "$PWD:/app" -v "${{ steps.composer-cache.outputs.dir }}:/tmp/cache" composer:2.1.11 install --no-interaction --no-progress # Run inside docker so composer install doesn't error...
```
Comment (eugene-manuilov) [inline .github/workflows/php-tests-wp-4-7-php-5-6.yml:74]: Looks like the newest version of Composer (released ~5 hours ago) contains backward-incompatible changes that cause a fatal error:

image

So, I have to update this command to make sure we use the latest compatible version of the `compose` image.

File: includes/Modules/Analytics.php [hunk 2]
```diff
@@ -1343,4 +1345,21 @@ protected function find_property( array $properties, $property_id = '', array $u
...
+	/**
+	 * Adds mode=analytics-step to the proxy params if the serviceSetupV2 feature flag is enabled.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $params An array of Google Proxy setup URL parameters.
+	 * @return array Updated array with the mode=analytics-step parameter.
+	 */
+	private function update_propxy_setup_mode( $params ) {
+		if ( Feature_Flags::enabled( 'serviceSetupV2' ) ) {
```
Comment (felixarntz) [inline includes/Modules/Analytics.php:1358]: I believe this condition also needs to include the check for whether the module is not connected (see ACs). If the module is already connected, the `mode=analytics-step` should not be included.

File: includes/Modules/Analytics.php [hunk 3]
```diff
@@ -1343,4 +1345,21 @@ protected function find_property( array $properties, $property_id = '', array $u
...
+
+	/**
+	 * Adds mode=analytics-step to the proxy params if the serviceSetupV2 feature flag is enabled.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $params An array of Google Proxy setup URL parameters.
+	 * @return array Updated array with the mode=analytics-step parameter.
+	 */
+	private function update_propxy_setup_mode( $params ) {
```
Comment (felixarntz) [inline includes/Modules/Analytics.php:1357]: Typo here.

**Review by felixarntz — CHANGES_REQUESTED** [review]
@eugene-manuilov Two small things to address here, then it will be good to merge.

Aside: This will eventually be adjusted as part of #2044 - where a new filter in `Google_Proxy::get_metadata_fields()` should be used instead of the `googlesitekit_proxy_setup_url_params` filter (which should then be dropped).

File: includes/Modules/Analytics.php (hunk 3 above)
Comment (eugene-manuilov) [reply includes/Modules/Analytics.php]: Updated.

File: includes/Modules/Analytics.php (hunk 2 above)
Comment (eugene-manuilov) [reply includes/Modules/Analytics.php]: Good catch! Updated!

**Review by felixarntz — APPROVED** [review]
Great!

**Pull request merged**